etapa1/parser.y

```diff
@@ -0,0 +1,70 @@
+%{
+#include<stdio.h>
+#define YYERROR_VERBOSE 1
+extern int yylineno;
+int yylex(void);
+void yyerror (char const *s);
+%}
+
+%token TK_PR_INT
+%token TK_PR_FLOAT
+%token TK_PR_BOOL
+%token TK_PR_CHAR
+%token TK_PR_STRING
+%token TK_PR_IF
+%token TK_PR_THEN
+%token TK_PR_ELSE
+%token TK_PR_WHILE
+%token TK_PR_DO
+%token TK_PR_INPUT
+%token TK_PR_OUTPUT
+%token TK_PR_RETURN
+%token TK_PR_CONST
+%token TK_PR_STATIC
+%token TK_PR_FOREACH
+%token TK_PR_FOR
+%token TK_PR_SWITCH
+%token TK_PR_CASE
+%token TK_PR_BREAK
+%token TK_PR_CONTINUE
+%token TK_PR_CLASS
+%token TK_PR_PRIVATE
+%token TK_PR_PUBLIC
+%token TK_PR_PROTECTED
+%token TK_PR_END
+%token TK_PR_DEFAULT
+%token TK_OC_LE
+%token TK_OC_GE
+%token TK_OC_EQ
+%token TK_OC_NE
+%token TK_OC_AND
+%token TK_OC_OR
+%token TK_OC_SL
+%token TK_OC_SR
+%token TK_LIT_INT
+%token TK_LIT_FLOAT
+%token TK_LIT_FALSE
+%token TK_LIT_TRUE
+%token TK_LIT_CHAR
+%token TK_LIT_STRING
+%token TK_IDENTIFICADOR
+%token TOKEN_ERRO
+%start programa
+
+%%
+
+// programa : funcoes;
+programa: variaveis;
+// programa: funcoes variaveis;
+programa: ;
+
+variaveis : tipo list_id;
+list_id : TK_IDENTIFICADOR | list_id',';
+tipo : TK_PR_BOOL | TK_PR_INT;
+
+%%
+
+void yyerror(const char *str)
+{
+    fprintf(stderr,"error: %s in line %d\n", str, yylineno);
+}
```

Comment:
https://stackoverflow.com/questions/62115979/how-to-implement-better-error-messages-for-flex-bison

da pra printar a coluna
acho bonito
mas fiquei com preguiça